api/Makefile

```diff
@@ -3,7 +3,8 @@
 all: clean test build
 
 test:
-	GOLANG_ENV=test go test -coverprofile=coverage.txt -covermode=atomic -cover $(shell go list ./... | grep -v /vendor/)
+	GOLANG_ENV=test go test -coverpkg=./... -coverprofile=coverage-integration.txt -covermode=atomic -cover github.com/18F/e-QIP-prototype/api/integration
+	GOLANG_ENV=test go test -coverprofile=coverage.txt -covermode=atomic -cover $(shell go list ./... | grep -v /vendor/ | grep -v integration)
 
```

Comment:
I think this is fine. I'm not sure I care about splitting out the integration tests from the rest, how much of the code is covered by all the tests seems like the metric we care about. 

I would also exclude /cmd/ from the list. Not sure that it makes a huge difference LOC-wise but I don't think we care about testing them the way we do the rest. This would be: 

`go test -coverprofile=coverage.txt -covermode=atomic -coverpkg=$(shell go list ./... | grep -v /vendor/ | grep -v /cmd | tr '\n' ',') -cover $(shell go list ./... | grep -v /vendor/)`

It's fine as is, I don't expect this to change the numbers much, and, in fact, my change pollutes the output of go test rather horribly with all the comma separated packages. It might be worth separating out the two runs even though that increases the amount of time spent testing.

Reply:
Thanks. I left `cmd` in because a handful of those commands are now officially part of a production build/usage. We need to tease out officially in documentation (and perhaps testing rigor) which ones are just dev/test tools and which aren't.

The `vendor` accommodation I left in for now but could hypothetically be removed if no downstream efforts are vendor'ing.

I had originally lumped the integration tests with the package tests. The overall coverage metric didn't change, but the per package metrics were all skewed. The overall coverage is what we care about by letter of the QASP, but it seemed good per-package metrics helped identify where we might want to spend our time productively (though it's only one level up from the code files).